dhcpv4/client.go

```diff
@@ -118,6 +138,31 @@ func MakeListeningSocket(ifname string) (int, error) {
 	return fd, nil
 }
 
+func (c *Client) getUDPAddresses() (*net.UDPAddr, *net.UDPAddr, error) {
+	var laddr *net.UDPAddr
+	if c.LocalAddr == nil {
+		laddr = &net.UDPAddr{IP: net.IPv4bcast, Port: ServerPort}
+	} else {
```

Comment:
I think this function should also check that the IP addresses are a valid IPv4, using `addr.To4()` and checking if not `nil`. This would avoid any risk in the functions that use it.

nit: to avoid duplication it may even be worth to call this function `toUDPAddr(addr net.Addr) (*net.UDPAddr, error)` and call it once for LocalAddr and once for RemoteAddr, but feel free to leave it as it is

Reply:
Changing it to `toUDPAddr` also makes confusing the returned values impossible, which I like a lot. I'll change this.

Reply:
I actually made the signature `toUDPAddr(addr net.Addr, defaultAddr *net.UDPAddr)  (*net.UDPAddr, error)` so we can pass a default value to use if `addr` is `nil` (instead of returning `nil` and then having to check for that in the caller).